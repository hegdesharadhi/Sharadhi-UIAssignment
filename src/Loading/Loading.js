import React, { useState, useEffect } from 'react'
import { buildInfo } from '../utils'
import fetch from '../api/data'
import Table from '../Table/Table'

const Loading = () => {
  const [data, setData] = useState(null) // Setting the initial value to null
  useEffect(() => {
    try {
      fetch().then((data) => {
        const info = buildInfo(data)
        setData(info)
      })
    } catch (error) {
      console.log(error)
    }
  }, []) // On the initial load, setting the value of data

  if (data == null) {
    return <div>Loading...</div>
  }
  return (
    <div>
      <Table data={data.summaryByCustomer} />
    </div>
  )
}

export default Loading
