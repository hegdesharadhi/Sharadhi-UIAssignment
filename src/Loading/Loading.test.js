import { render, screen } from '@testing-library/react'
import { data } from './mockData'
import Loading from './Loading'

describe('Loading', () => {
  test('Check the Loading when data is empty', () => {
    render(<Loading data={data} />)
    const loading = screen.getByText(/Loading/i)
    expect(loading).toMatchSnapshot()
  })
})
