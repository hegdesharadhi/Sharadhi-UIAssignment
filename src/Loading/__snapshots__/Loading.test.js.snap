// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Check the Loading when data is empty 1`] = `
<div>
  Loading...
</div>
`;
